_adr/87/index.adoc

```diff
@@ -1,48 +1,227 @@
+
+KAS Fleet Manager controls the labels exposed for each metric returned in all endpoints. 
+Metrics retrieved from Observatorium are also filtered so that end users will only receive metrics for their own Kafka instance.
+Apart from these, no additional aggregation, filtering or modification of the metric is done by KAS Fleet Manager.
 
 ## Proposal
-// What is the decision being proposed
+### Metric and Label Naming
+All metric and label names must adhere to the policy stated in link:../../_ap/16/index.adoc[AP-16: Naming Policy for Customer Facing Metrics].
+
+### Kafka Metric Categories
+The customer facing Kafka metrics can be divided into multiple categories.
+Categorizing these metrics into logical groups can help us formulate the metric name and document it.
+
+|===
+|Category | Description | Sub-component (to be specified in the metric name)
+
+|Instance
+|General Kafka metrics not specific to any other sub-components.
+|instance
+
+|Broker
+|Metrics associated to Kafka brokers
+|broker
+
+|Topic
+|Metrics associated to Kafka topics
+|topic (Even though the topic metrics already have a 'topic' label, including this improves the clarity of the metric name)
+|===
+
+Examples of sub-component usage in the metric name:
+
+* For *Instance* metrics, the metric prefix would be `kas_instance_`.
+* For *Broker* metrics, the metric prefix would be `kas_broker_`.
+* For *Topic* metrics, the metric prefix would be `kas_topic_`.
+
+#### Labels
+At minimum, each metric should have a label that identifies the Kafka instance if possible.
+This would be useful if a user has multiple Kafka instances.
+If they want to integrate all of their Kafka metrics into one monitoring system, the label(s) can act as an identifier.
+
+### Process for requesting new customer facing metric
+When requesting for a new metric to be exposed in the KAS Fleet Manager API, a new issue in JIRA or Github must be created.
+The description of this issue should have sufficient information on what this new metric should provide in order to identify the metric to be used and how it should be documented in later stages.
+
+The issue created must have the following sub-tasks and they must be completed in order:
+
+. Identify the metric and labels to be used and how it will be aggregated if required.
+** This should be assigned to the team that has the knowledge of the requested metric.
+** [.underline]#Definition of Done#:
+*** Metric to be used is identified.
+*** (Optional) Aggregation of the metric determined.
+*** (Optional) Category/Sub-component which this new metric belongs to is identified.
```

Comment:
This is the only place where we mention Category/Sub-component, elsewhere in the two documents, it is only mentioning Sub-component. I think we should keep it homogeneous.

This leads to my more general question. Is sub-component a correct model? I understand we don't mention the technology names like Strimzi, still the sub-components are possible implementation details of the service. But by exposing them, we are making them aprt of the customer contract and thus be documented. So instance, brokers and topics are concepts we want exposed in the API?

If yes, that's good but in the AP and ADRs, I would still add a caution block explaining ththat we do not want to expose things we consider implementation details that we would reserve the right to change later as names for sub-components.

Reply:
Good point. I think topics and brokers are foundational concepts in a Kafka service, so it's impossible to avoid the manifesting in this API. But we should be very careful about things which are not intrinsic to the service being provided. For example, metrics from other components. Those will often be _about_ the foundational concepts (like topics and brokers) but their _origin_ (say, canary or kafka exporter) is an implementation detail which shouldn't be exposed in the API.

Reply:
> This is the only place where we mention Category/Sub-component, elsewhere in the two documents, it is only mentioning Sub-component. I think we should keep it homogeneous.

Would category make more sense in terms of terminology than a sub-component? Category is what we've been using to group these metrics in our documentation. Topics/brokers would probably makes sense as a 'category' rather than a 'sub-component' too. 

> If yes, that's good but in the AP and ADRs, I would still add a caution block explaining ththat we do not want to expose things we consider implementation details that we would reserve the right to change later as names for sub-components.

> Those will often be about the foundational concepts (like topics and brokers) but their origin (say, canary or kafka exporter) is an implementation detail which shouldn't be exposed in the API.

Good point, i'll add a caution block in these sections to make sure we do not expose implementation details.

Reply:
Category intuitively works for me but I haven't played much with the concepts.